### ISSUE TITLE
Support sorting by "GET /resources" endpoint

### ISSUE THREAD

**Issue by maxceem** [topcoder-platform/resources-api#51]
We have a feature request for Admin App to be able to sort challenge resources by:
- `memberHandle`
- `created`
- Is it hard to support sorting by `role text` alphabetically: `Registered`, `Submitter`, `Observer`?

@ThomasKranitsas @rootelement

**Comment by rootelement** [comment]
Merged this:
https://github.com/topcoder-platform/resources-api/pull/54
@maxceem please test in dev

**Comment by maxceem** [comment]
@rootelement deployment of this PR failed https://app.circleci.com/pipelines/github/topcoder-platform/resources-api/183/workflows/19fb405b-8662-422d-9f97-6b2de586a546/jobs/156

**Comment by maxceem** [comment]
@rootelement `"sortBy" is required` now, though it feels that it should be optional.
It's also marked as optional in Swagger

![image](https://user-images.githubusercontent.com/146016/103012486-5e78cd80-4544-11eb-8695-e03cea6ec151.png)

but in code it's marked as required:

![image](https://user-images.githubusercontent.com/146016/103012558-73556100-4544-11eb-9bcc-cbddd1ab1aa4.png)

**Comment by maxceem** [comment]
@rootelement now sorting params are optional and sorting by `created` looks good. 
But spring by `memebrHandle` doesn't return any results, while the same request without sorting returns data.

For example, request with sorting by `memberHandle` returns nothing:
```
curl 'https://api.topcoder-dev.com/v5/resources?challengeId=bcac639d-9d56-458b-bbbf-efc13b1e342a&page=1&perPage=100&sortBy=memberHandle&sortOrder=asc' \
  -H 'authority: api.topcoder-dev.com' \
  -H 'accept: application/json' \
  -H 'access-control-expose-headers: *' \
  -H 'authorization: Bearer <TOKEN>'
```

while the same request without sorting returns records:
```
curl 'https://api.topcoder-dev.com/v5/resources?challengeId=bcac639d-9d56-458b-bbbf-efc13b1e342a&page=1&perPage=100' \
  -H 'authority: api.topcoder-dev.com' \
  -H 'accept: application/json' \
  -H 'access-control-expose-headers: *' \
  -H 'authorization: Bearer <TOKEN>'
```

**Comment by maxceem** [comment]
@rootelement at the moment no resources are returned by this endpoint even without sorting params for any challenge.

**Comment by maxceem** [comment]
@ThomasKranitsas at the moment Resouces API doesn't return any resources with or without scoring params. It always returns 0 resources for all the challenges on DEV.

**Comment by ThomasKranitsas** [comment]
Hey @maxceem, seems like there was a change on the API in order to support sorting but in order for this to work we need to update the mappings in ES and reindex the data. I have temporarily reset this change so the API works again.

cc @rootelement

**Comment by maxceem** [comment]
Thanks, @ThomasKranitsas. Now Resource API returns data for me as before.

Though I can still reproduce this issue for some challenges.
For example, I've created a new challenge, and assigned a copilot https://challenges.topcoder-dev.com/projects/16668/challenges/92af11c5-df2e-4703-8d32-f4b80a5dc064/edit
When I get the list of resources https://api.topcoder-dev.com/v5/resources?challengeId=92af11c5-df2e-4703-8d32-f4b80a5dc064 I get 0 results. Though I've already created one copilot record.
And if I try to assign a copilot again I get an error that the resource is already there: `"User max.admin already has resource with roleId: cfe12b3f-2a24-4639-9d8b-ec86726f76bd in challenge: 92af11c5-df2e-4703-8d32-f4b80a5dc064"`

I'm not sure if this is connected with the issue, or should I log a new issue for this.

**Comment by ThomasKranitsas** [comment]
No, no need for a new issue. Seems like a recent change broke something.. I'll investigate and fix

**Comment by ThomasKranitsas** [comment]
This has been fixed

**Comment by rootelement** [comment]
@maxceem is this unblocked now? Can you finish the admin in dev?

**Comment by maxceem** [comment]
@rootelement this is not fixed yet.

When I get resources without sorting it works:

![image](https://user-images.githubusercontent.com/146016/104333283-baa69000-54f9-11eb-89ca-b933185b431c.png)

But when I apply sorting no results are returned:

![image](https://user-images.githubusercontent.com/146016/104333340-cc883300-54f9-11eb-8d6f-84d2b9d72ba1.png)

**Comment by ThomasKranitsas** [comment]
`memberHandle` is not yet supported as we need to reindex the existing data.

cc @rootelement
